controllers/hetznercluster_controller.go

```diff
@@ -173,6 +173,24 @@ func (r *HetznerClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 func (r *HetznerClusterReconciler) reconcileNormal(ctx context.Context, clusterScope *scope.ClusterScope) (ctrl.Result, error) {
 	hetznerCluster := clusterScope.HetznerCluster
 
+	// write ssh key name from secret to spec of HetznerCluster if it is specified
+	sshKeyName, ok := clusterScope.HetznerSecret().Data[hetznerCluster.Spec.HetznerSecret.Key.SSHKey]
```

Comment:
What should we do, if sshKeyName exists, but it is the empty string?

Reply:
yeah I thought about it. Probably it makes sense to also add that check